### PR TITLE
Fix time.Time not equals bug

### LINKED ISSUE #950: assert.ObjectsAreEqual work not correct with serialized\deserialized time.Time
**time.Time** not equals before and after serialization/deserialization to JSON on machine in **UTC**
reason: https://github.com/golang/go/issues/19486#issuecomment-292610767

code: https://play.golang.org/p/H0ACWIBZE3O

solution: change the **time.Time** comparison format. For example, compare time using **time.Equal**

### PR DESCRIPTION
**Pull request by senyast4745**
Adding test

## Summary
Fix assert.ObjectsAreEqual bug with not equals time.Time before and after serializing\deserializing

## Changes
Added a check in ObectsAreEqual in case the arguments are a time object

## Motivation
Fix bug to use test in docker 

## Related issues
Closes #950

### CODE REVIEW

File: assert/assertions_test.go [hunk 1]
```diff
@@ -153,6 +153,21 @@ func TestObjectsAreEqual(t *testing.T) {
 
 }
 
+func TestObjectsAreEqual_Time(t *testing.T) {
```
Comment (boyan-soubachov) [inline assert/assertions_test.go:156]: Should we not have the test name explicitly state that we're testing serialized/json time representations? i.e. `TestObjectsAreEqual_SerializedTime`?

File: assert/assertions.go [hunk 2]
```diff
@@ -60,6 +60,15 @@ func ObjectsAreEqual(expected, actual interface{}) bool {
 		return expected == actual
 	}
 
+	tExp, ok := expected.(time.Time)
+	if ok {
```
[the remark this answers is not in the document]
Comment (senyast4745) [reply assert/assertions.go]: Done

File: assert/assertions_test.go (hunk 1 above)
Comment (senyast4745) [reply assert/assertions_test.go]: Done

**Comment by boyan-soubachov** [comment]
I've had to fix the TravisCI config. Do you mind rebasing the branch and then I can merge it?

**Comment by senyast4745** [comment]
Of course I do not mind

**Comment by senyast4745** [comment]
I can merge the master in my branch pulling up new changes

**Comment by boyan-soubachov** [comment]
Thank you, there still seem to be some conflicts, do you mind having a look again?

**Comment by senyast4745** [comment]
Sorry, upstream has not been updated. The last commit pulled all the changes from testify

**Comment by senyast4745** [comment]
Thanks in advance for the opportunity to contribute))

**Comment by boyan-soubachov** [comment]
Something is very weird, still getting errors due to conflicts:
image

Are you able to rebase and squash maybe? I promise I will merge it as soon as you do :)

**Comment by senyast4745** [comment]
Hi, very sorry, I was very busy. I merge the latest changes from master to my branch. Unfortunately, the commits could not be merged, I'm afraid to do a hard push or rebase

**Comment by boyan-soubachov** [comment]
> Hi, very sorry, I was very busy. I merge the latest changes from master to my branch. Unfortunately, the commits could not be squashed, I'm afraid to do a hard push or rebase

If you rebase, resolve any conflicts and then force-push your branch you should be fine.

We can't merge this PR while there are conflicts :)

**Comment by senyast4745** [comment]
Please show me in which files the conflicts occur, I will solve them locally, because I still don't understand a little how and where I can rebase and force-push

**Comment by senyast4745** [comment]
should i rebase my branch to master?

**Comment by boyan-soubachov** [comment]
> should i rebase my branch to master?

Yes, you'd rebase on `master`, something along the lines of
```
git rebase master
<resolve conflicts with your editor>
git push -f fork
```

where `fork` is the remote set-up to your fork.

This is a good and thorough guide on the general flow: https://blog.scottlowe.org/2015/01/27/using-fork-branch-git-workflow/

**Comment by senyast4745** [comment]
I did rebase and force-push, as you asked

**Comment by leoleoasd** [comment]
Should we close this given that #979 fixed the git history issue?

**Pull request closed without merging**